torch/fx/experimental/proxy_tensor.py

```diff
@@ -491,7 +508,7 @@ def wrapped(*args):
         else:
             raise AssertionError(f"Unexpected tracing type: {tracing_mode}")
 
-        proxy_mode = ProxyTorchDispatchMode(fx_tracer) if trace_factory_functions else nullcontext()
+        proxy_mode = ProxyTorchDispatchMode(fx_tracer, trace_factory_functions=trace_factory_functions)
```

Comment:
nb: from a perf perspective this is worse than before, as you're forced to go into the torch dispatch hook even though you don't do anything with it. It doesn't matter too much though since trace factory functions is usually true.

Reply:
Well, now we always need to create a `ProxyTorchDispatchMode` to be attached to `ProxyTensor`. When trace_factory_functions is false we still need the dispatch mode and not a nullcontext to pass it to the `wrap_key` function.

Reply:
What breaks if we don't enable the mode (keep this line as it was previously), and don't have the trace_factory_functions argument? In my mind, if we don't have the mode on shouldn't it go:

Factory Function -> No Python Key (since no arguments have the python key and there's no mode) -> Not Traced

Non-factory function -> Python Key (from arguments) -> Mode -> redispatch on python key -> mode's torch dispatch

Reply:
`AttributeError: 'nullcontext' object has no attribute 'restore'` when called `with proxy_mode.restore()`. We can try to intercept this case and call `proxy_call` with `nullcontext` passed as mode. Then at least one test fails `TestProxyTensorCPU.test_mode_tracing_factory_function_no_factory_function_cpu` with 
`TypeError: unsupported operand type(s) for +: 'ProxyTensor' and 'Tensor'`.